pkg/controller/syncset/syncset_controller.go

```diff
@@ -0,0 +1,439 @@
+		return reconcile.Result{}, nil
+	}
+
+	origCD := cd
+	cd = cd.DeepCopy()
+
+	cdLog.Info("reconciling sync sets for cluster deployment")
+
+	// get all sync sets that apply to cd
+	syncSets, err := r.getRelatedSyncSets(cd)
+	if err != nil {
+		cdLog.WithError(err).Error("unable to list related sync sets for cluster deployment")
+		return reconcile.Result{}, err
+	}
+
+	// get all selector sync sets that apply to cd
+	// selectorSyncSets, err := r.getRelatedSelectorSyncSets(cd)
+	// if err != nil {
+	//      cdLog.WithError(err).Error("unable to list related sync sets for cluster deployment")
+	//  	return reconcile.Result{}, err
+	// }
+
+	// get kubeconfig for the cluster
+	secretName := cd.Status.AdminKubeconfigSecret.Name
+	secretData, err := r.loadSecretData(secretName, cd.Namespace, adminKubeConfigKey)
+	if err != nil {
+		cdLog.WithError(err).Error("unable to load admin kubeconfig")
+		return reconcile.Result{}, err
+	}
+	kubeConfig := []byte(secretData)
+
+	for _, syncSet := range syncSets {
+		ssLog := cdLog.WithFields(log.Fields{"syncSet": syncSet.Name})
+		ssLog.Debug("applying sync set")
+
+		syncSetStatus := findSyncSetStatus(syncSet.Name, cd.Status)
+		err = r.applySyncSet(syncSet, kubeConfig, &syncSetStatus, ssLog)
+		if err != nil {
+			ssLog.WithError(err).Error("unable to apply sync set")
+		}
+		cd.Status.SyncSetStatus = appendOrUpdateSyncSetObjectStatus(cd.Status.SyncSetStatus, syncSetStatus)
+	}
+
+	err = r.updateClusterDeploymentStatus(cd, origCD, cdLog)
+	if err != nil {
+		cdLog.WithError(err).Errorf("error updating cluster deployment status")
+		return reconcile.Result{}, err
+	}
+
+	cdLog.Info("done reconciling sync sets for cluster deployment")
+
+	return reconcile.Result{}, nil
+}
+
+// applySyncSet evaluates all resources and patches in a given SyncSet and applies them to the cluster identified by kubeConfig
+func (r *ReconcileSyncSet) applySyncSet(ss hivev1.SyncSet, kubeConfig []byte, syncSetStatus *hivev1.SyncSetObjectStatus, ssLog log.FieldLogger) error {
+	h := r.applierBuilder(kubeConfig, r.logger)
+
+	for _, resource := range ss.Spec.Resources {
+		info, err := h.Info(resource.Raw)
+		if err != nil {
+			syncSetStatus.Conditions = r.setUnknownObjectSyncCondition(syncSetStatus.Conditions, err)
+			return err
+		}
+
+		resourceHash := md5.Sum(resource.Raw)
+		resourceSyncStatus := hivev1.SyncStatus{
+			APIVersion: info.APIVersion,
+			Kind:       info.Kind,
+			Name:       info.Name,
+			Namespace:  info.Namespace,
+			Hash:       fmt.Sprintf("%x", resourceHash[:]),
+		}
+
+		// determine if resource has been created or is different based on the resource md5 sum
+		found := false
+		different := false
+		for _, rss := range syncSetStatus.Resources {
+			if rss.Name == resourceSyncStatus.Name && rss.Namespace == resourceSyncStatus.Namespace {
+				found = true
+				if rss.Hash != resourceSyncStatus.Hash {
+					different = true
+				}
+				break
+			}
+		}
+
+		if !found || different {
```

Comment:
The condition here should probably be a little bit more nuanced ... if it's not found, different or a sufficient amount of time has passed since the last condition probe time, then we should go ahead and apply it.